jsapp/js/components/processing/singleProcessing.tsx

```diff
@@ -0,0 +1,202 @@
+import React from 'react'
+import {RouteComponentProps} from 'react-router'
+import {actions} from 'js/actions'
+import {getTranslatedRowLabel} from 'js/assetUtils'
+import assetStore from 'js/assetStore'
+import bem, {makeBem} from 'js/bem'
+import {AnyRowTypeName} from 'js/constants'
+import LoadingSpinner from 'js/components/common/loadingSpinner'
+import SingleProcessingHeader from 'js/components/processing/singleProcessingHeader'
+import SingleProcessingSubmissionDetails from 'js/components/processing/singleProcessingSubmissionDetails'
+import SingleProcessingContent from 'js/components/processing/singleProcessingContent'
+import './singleProcessing.scss'
+
+bem.SingleProcessing = makeBem(null, 'single-processing', 'section')
+bem.SingleProcessing__top = makeBem(bem.SingleProcessing, 'top', 'section')
+bem.SingleProcessing__bottom = makeBem(bem.SingleProcessing, 'bottom', 'section')
+bem.SingleProcessing__bottomLeft = makeBem(bem.SingleProcessing, 'bottom-left', 'section')
+bem.SingleProcessing__bottomRight = makeBem(bem.SingleProcessing, 'bottom-right', 'section')
+
+/**
+ * this.props.params properties
+ */
+type SingleProcessingProps = RouteComponentProps<{
+  uid: string,
+  questionName: string,
+  submissionId: string,
+}, {}>
+
+type SingleProcessingState = {
+  isSubmissionCallDone: boolean
+  isIdsCallDone: boolean
+  submissionData: SubmissionResponse | null
+  submissionsIds: string[]
+  asset: AssetResponse | undefined
+  error: string | null
+}
+
+/**
+ * This route component is being loaded with PermProtectedRoute so we know that
+ * the call to backend to get asset was already made :happy_face:
+ */
+export default class SingleProcessing extends React.Component<SingleProcessingProps, SingleProcessingState> {
+  constructor(props: SingleProcessingProps) {
+    super(props)
+    this.state = {
+      isSubmissionCallDone: false,
+      isIdsCallDone: false,
+      submissionData: null,
+      submissionsIds: [],
+      asset: assetStore.getAsset(this.props.params.uid),
+      error: null,
+    }
+  }
+
+  private unlisteners: Function[] = []
+
+  componentDidMount() {
+    this.unlisteners.push(
+      actions.submissions.getSubmission.completed.listen(this.onGetSubmissionCompleted.bind(this)),
+      actions.submissions.getSubmission.failed.listen(this.onGetSubmissionFailed.bind(this)),
+      actions.submissions.getSubmissionsIds.completed.listen(this.onGetSubmissionsIdsCompleted.bind(this)),
+      actions.submissions.getSubmissionsIds.failed.listen(this.onGetSubmissionsIdsFailed.bind(this)),
+    )
+    actions.submissions.getSubmission(this.props.params.uid, this.props.params.submissionId)
+    actions.submissions.getSubmissionsIds(this.props.params.uid)
+  }
+
+  componentDidUpdate(prevProps: any) {
```

Comment:
You're right, this should be `SubmissionProcessingProps`. As for `this.props.params` instead of just `this.props` - this is happening because we use `RouteComponentProps` instead of simple type object. And we use this because this is a component used for rendering route